src/tests/services/configuration.test.ts

```diff
@@ -151,14 +178,51 @@ suite('Test configurations', () => {
         assert.ok(config.canSend(event) === true, `${event.event} should be sent`);
 
         event = {
-            userId: "533629ec-091b-474b-95e6-3aa0eef3e940",//numvalue:0.22 < (1- 0.7)
+            userId: "72f09a0e-1fa6-46d1-8322-48ac0ffa4252",//numvalue:0.189 < (1- 0.7)
+            event: "verbose-event",
+        } as AnalyticsEvent;
+        assert.ok(config.canSend(event) === true, `${event.event} should be sent`);
+
+
+        event = {
+            userId: "ddf95114-333d-41e0-b1ba-d84bc6293634",//numvalue:0.3579 > (1 - 0.7)
+            event: "verbose-event",
+        } as AnalyticsEvent;
+        assert.ok(config.canSend(event) === false, `${event.event} should not be sent`);
+    });
+
+
+    test('Should exclude 99.7% of event', async () => {
+    // Generate test UUIDs
+    // const uuid = '';
+    // var index = 0;
+    // while (true) {
+    //   index++;
+    //   const uuid = generateUUID();
+    //   const num = numValue(uuid);
+    //   if (num < 0.003) {
```

Comment:
Just something I noticed while playing with this. The `canSend(..)` function for an `extremeRatioedEvent` configuration (exclude 0.997) would send a uuid whose `numValue` is 0.003 . So this should be `num <= 0.003` shouldn't it (for the sake of accuracy) ?

At least based on the logic in https://github.com/redhat-developer/vscode-redhat-telemetry/blob/2cc060d21518eee86d7fa20c40a57e2abfab55c9/src/common/impl/configuration.ts#L102-L104 .